tests/optimize/curve_fit_test.py

```diff
@@ -0,0 +1,119 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2022 Scipp contributors (https://github.com/scipp)
+import numpy as np
+import scipp as sc
+from scipp.optimize import curve_fit
+
+import pytest
+
+
+def func(x, a, b):
+    return a * sc.exp(-(b / x.unit) * x)
+
+
+def array1d(*, a=1.2, b=1.3, noise_scale=0.1):
+    size = 50
+    x = sc.linspace(dim='xx', start=-0.1, stop=4.0, num=size, unit='m')
+    y = func(x, a, b)
+    rng = np.random.default_rng()
+    # Noise is random but avoiding unbounded values to avoid flaky tests
+    y.values += noise_scale * np.clip(rng.normal(size=size), -2.0, 2.0)
+    return sc.DataArray(y, coords={'xx': x})
+
+
+def test_should_not_raise_given_function_with_dimensionless_params_and_1d_array():
+    curve_fit(func, array1d())
+
+
+def test_should_raise_TypeError_when_xdata_given_as_param():
+    with pytest.raises(TypeError):
+        curve_fit(func, array1d(), xdata=np.arange(4))
+
+
+def test_should_raise_TypeError_when_ydata_given_as_param():
+    with pytest.raises(TypeError):
+        curve_fit(func, array1d(), ydata=np.arange(4))
+
+
+def test_should_raise_TypeError_when_sigma_given_as_param():
+    with pytest.raises(TypeError):
+        curve_fit(func, array1d(), sigma=np.arange(4))
+
+
+def test_should_raise_NotFoundError_when_data_array_has_no_coord():
```

Comment:
What about multiple coords, multi-dim coord, multi-dim data?

Reply:
Added a test checking the DimensionError is raised if data is multi-dim. Since this is not supported there is nothing to test with multiple coords or multi-dim coords.

Reply:
This is basically what I meant.